etcdserver/raft.go

```diff
@@ -108,10 +108,18 @@ type raftNode struct {
 	done    chan struct{}
 }
 
+func (r *raftNode) start(s *EtcdServer) {
```

Comment:
can we add a comment about start? also is it possible to combine start and run? Ideally we should get rid of the passed in server structure.